### PR TITLE
chore: update github issue templates

### PR DESCRIPTION
**Pull request by sergiught**
## Description

In order to improve our GitHub support workflow, this PR introduces [issue forms](https://docs.github.com/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms), a new type of issue template.

We have been using these for an extensive amount of time within Auth0 SDKs & CDT repos and our data shows that this increases the quality of received GitHub issues and also makes them easier to fill out for reporters. 

## References

image
image

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+blank_issues_enabled: false
```
Comment (sergiught) [inline .github/ISSUE_TEMPLATE/config.yml:1]: This disables the ability to create blank issues. With the changes in this PR only bug reports or feature requests will be opened. For any kind of other more loose discussions we're attaching here a link to the Discussions page.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: 🚀 Feature Request
+description: Suggest an idea or a feature for this project
+labels: [ ":seedling: feature" ]
```
Comment (sergiught) [inline .github/ISSUE_TEMPLATE/feature_request.yml:3]: This is a proposal for a new label to be created before this PR gets merged.

File: .github/ISSUE_TEMPLATE/report_a_bug.yml [hunk 3]
```diff
@@ -0,0 +1,144 @@
+name: 🐛 Report a bug
+description: Have you found a bug or issue? Create a bug report for OpenFGA
+labels: [ ":beetle: bug" ]
```
Comment (sergiught) [inline .github/ISSUE_TEMPLATE/report_a_bug.yml:3]: This is a proposal for a new label to be created before this PR gets merged.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 2 above)
Comment (rhamzeh) [reply .github/ISSUE_TEMPLATE/feature_request.yml:3]: We have `enhancement` and `bug` currently, can we use those instead?

Comment (rhamzeh) [reply .github/ISSUE_TEMPLATE/feature_request.yml:3]: Also... I have a deep animosity to emojis in commits & code, so can we do a compromise and keep the emoji in the title but not in the label? :)

Comment (sergiught) [reply .github/ISSUE_TEMPLATE/feature_request.yml:3]: @rhamzeh Absolutely, we don't really need to create new ones, we can reuse current ones, or even update them. 

I'm just biased towards using emojis 😄 , here's an example in action: https://github.com/auth0/terraform-provider-auth0/labels. 

I can revert the label at any time, just waiting for further feedback to apply more changes in bulk.

File: .github/ISSUE_TEMPLATE/report_a_bug.yml (hunk 3 above)
Comment (sergiught) [reply .github/ISSUE_TEMPLATE/report_a_bug.yml]: Removed in [`8b59dc8` (#1262)](https://github.com/openfga/openfga/pull/1262/commits/8b59dc8d3ff120b7d0e8cb521b4101cfc5e90df8).

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 2 above)
Comment (sergiught) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: Removed in [`8b59dc8` (#1262)](https://github.com/openfga/openfga/pull/1262/commits/8b59dc8d3ff120b7d0e8cb521b4101cfc5e90df8).

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+contact_links:
+  - name: 📖 OpenFGA's Documentation
+    url: https://openfga.dev/docs/authorization-and-openfga
```
Comment (adriantam) [inline .github/ISSUE_TEMPLATE/config.yml:4]:
```suggestion
    url: https://openfga.dev/
```
Should link to top page in case we add/remove/rename pages

Comment (sergiught) [reply .github/ISSUE_TEMPLATE/config.yml]: Updated in [`87631cc` (#1262)](https://github.com/openfga/openfga/pull/1262/commits/87631cc931b355ce7fd557f34e7250ecbc3d5119)

Left the `/docs` as it's supposed to go directly to the docs page and not the home page.

**Review by rhamzeh — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+blank_issues_enabled: false
+contact_links:
+  - name: 📖 OpenFGA's Documentation
+    url: https://openfga.dev/docs
+    about: Check OpenFGA's documentation for an in-depth overview
+  - name: 👽 Discord
+    url: https://discord.gg/8naAwJfWN6
+    about: Join OpenFGA's active Discord community
+  - name: 📝 RFC's
```
Comment (rhamzeh) [inline .github/ISSUE_TEMPLATE/config.yml:9]:
```suggestion
  - name: 📝 RFCs
```

Comment (rhamzeh) [reply .github/ISSUE_TEMPLATE/config.yml:9]: Sorry missed this earlier 🙈

**Review by rhamzeh — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/config.yml (hunk 5 above)
Comment (sergiught) [reply .github/ISSUE_TEMPLATE/config.yml]: All good @rhamzeh ⭐, appreciate the keen eye 👁️

Comment (sergiught) [reply .github/ISSUE_TEMPLATE/config.yml]: Accounted for within [`47b46a0` (#1262)](https://github.com/openfga/openfga/pull/1262/commits/47b46a0b2c747db9857ae2fc9e75b8bcb4364b66)

**Review by rhamzeh — APPROVED** [review]

**Review by rhamzeh — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/report_a_bug.yml [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        - label: I have looked into the [README](https://github.com/openfga/openfga/blob/main/README.md) and have not found a suitable solution or answer.
+          required: true
+        - label: I have looked into the [documentation](https://openfga.dev/docs) and have not found a suitable solution or answer.
+          required: true
+        - label: I have searched the [issues](https://github.com/openfga/openfga/issues) and have not found a suitable solution or answer.
+          required: true
+        - label: I have upgraded to the [latest version](https://github.com/openfga/openfga/releases/latest) of OpenFGA and the issue still persists.
+          required: true
+        - label: I have searched the [Discord](https://discord.gg/8naAwJfWN6) community and have not found a suitable solution or answer.
+          required: true
```
Comment (miparnisari) [inline .github/ISSUE_TEMPLATE/report_a_bug.yml:30]: not a blocker for this PR, but I wouldn't like to encourage the community to go through Discord. That's _a lot_ of messages.

**Review by miparnisari — APPROVED** [review]

**Pull request merged**